modules/dxp/apps/analytics/analytics-reports-journal-impl/src/main/java/com/liferay/analytics/reports/journal/internal/info/item/JournalArticleAnalyticsReportsInfoItem.java

```diff
@@ -118,6 +126,77 @@ public WebImage getAuthorWebImage(
 		);
 	}
 
+	@Override
+	public String getCanonicalURL(
+		JournalArticle journalArticle, Locale locale) {
+
+		Optional<ThemeDisplay> themeDisplayOptional =
+			_getThemeDisplayOptional();
+
+		if (!themeDisplayOptional.isPresent()) {
+			return StringPool.BLANK;
+		}
+
+		return themeDisplayOptional.map(
+			themeDisplay -> {
+				Optional<Layout> layoutOptional = _getLayoutOptional(
+					journalArticle);
+
+				return layoutOptional.map(
+					layout -> {
+						HttpServletRequest httpServletRequest =
+							themeDisplay.getRequest();
+
+						LayoutDisplayPageObjectProvider
```

Comment:
Should it be parametrized? (even with a wildcard)

Reply:
What do you think should be parametrized?

Reply:
The LayoutDisplayPageObjectProvider. Something like:

```java
LayoutDisplayPageObjectProvider<?>
							initialLayoutDisplayPageObjectProvider =
								(LayoutDisplayPageObjectProvider<?>)
									httpServletRequest.getAttribute(
										LayoutDisplayPageWebKeys.
											LAYOUT_DISPLAY_PAGE_OBJECT_PROVIDER);
```

Reply:
Why do you want to parametrized it?

Reply:
Essentially, because raw types behaves just like they were before generics were introduced. So, you make sure the use of the parameter you want to use. 
 In this particular case, in which It's not a collection but just one object it is not so important. But generally it's a good practice to avoid it https://www.informit.com/articles/article.aspx?p=2861454

Reply:
yeahh!!, but I am asking why do you want to parametrize this object in this particular case

Reply:
As I said in this particular case it is not important, It's just a way to prevent possible undesired behaviors. But It's ok for me as it is ;)

Reply:
Ok, I was just curious, I could understand why would you want to parametrize the method _get LayoutDisplayPageObjectProvider (because this has a type that could affect the result).

I wanted to know why do you think it was so important in this case, and why it was not important for the _getLayoutDisplayPageObjectProvider method.

Reply:
Well, my only reason was that I missed this one XD